contracts/gateway/EIP20Gateway.sol

```diff
@@ -1084,13 +1075,14 @@ contract EIP20Gateway is GatewayBase {
         //Get the stake amount.
         stakeAmount_ = stakes[_messageHash].amount;
 
+        uint256 stakedBounty = stakes[_messageHash].bounty;
+
+        delete stakes[_messageHash];
```

Comment:
This deletion does not prevent the following `token.transfer(address(stakeVault), stakeAmount_);` in case of reentrancy. Problem?